### DISCUSSION TITLE
float problem

### DISCUSSION THREAD

**Discussion by zahidhasan83** [idaholab/moose#26396] [Q&A Modules: Phase field]
Dear MOOSE users,
Hope you are doing great! I tried to determine a material property. But the following error appears: 
```
 "cannot convert field 'Materials/consts_expected/prop_values' value 'Kn*An*(exp(-Eact/(Kb*T)))*(exp(-Qgb/(Kb*T)))' to float"

The input file structure looks like that:
[Materials]
  [./consts_expected]
    type = GenericConstantMaterial
    prop_names =  'Kn      An     Eact     Qgb    Kb    T             N'
    prop_values = '1.11e8    10e4    4.8   1   8.6173303e-5    800   Kn*An*(exp(-Eact/(Kb*T)))*(exp(-Qgb/(Kb*T)))'
    outputs = exodus
  [../]
```
Any suggestion what is wrong? Any solution?
I would appreciate if you can manage time  to reply.

Thanks,
Zahid

**Comment by GiudGiud** [comment]
hello

you need to parse the quantity down to a real before it gets fed into the material. You can get the parser to do that with:
```
Kn = 1.11e8
An = 10e4
Eact = 4.8
Qgb = 1
T =  8.6173303e-5
N = 800

[Materials]
  [./consts_expected]
    type = GenericConstantMaterial
    prop_names =  'Kn      An     Eact     Qgb    Kb    T             N'
    prop_values = '${Kn}    ${An}    ${Eact}   ${Qgb}  ${T}    ${N}  ${fparse Kn*An*(exp(-Eact/(Kb*T)))*(exp(-Qgb/(Kb*T)))}'
    outputs = exodus
  [../]
```

**Reply by GiudGiud** [reply]
what I m doing here is that i m defining the quantities as input file variables, then using the parser to convert the whole expression into a Real number.

The ${N} are doing a substitution, a simple one with no other operation.
The ${fparse is doing a parsed calculation and a substitution

**Reply by zahidhasan83** [reply]
@GiudGiud Thanks. I appreciate.
